spec/lib/goodcity/detail_factory_spec.rb

```diff
@@ -0,0 +1,14 @@
+require "rails_helper"
+require "goodcity/detail_factory"
+
+describe Goodcity::DetailFactory do
+  let(:item_params) {  }
+  let(:pacakge) {}
+  let(:detail_factory) { described_class.new(item_params, package) }
+
+  before do
+    stub_request(:get, /goodcitystorage.blob.core.windows.net/).
+      with(headers: { "Accept" => "*/*", "User-Agent" => "Ruby" }).
+      to_return(status: 200, body: file, headers: {}).response.body
+  end
```

Comment:
@abhinav098 could you add a few tests that show the various detail being added. Perhaps one approach would be to stub the method `Stockit::ItemSync.index_with_detail` to return a hash of detail params (rather than stubbing a web request) and confirm that they are applied to the package in GoodCity. The tests can confirm that each parameter is successfully applied.

Reply:
Sure, @steveyken , I'll work on this, once i'm done with the box and pallet addition/removal of items thingy.